src/components/ExpensiTextInput/BaseExpensiTextInput.js

```diff
@@ -118,51 +119,57 @@ class BaseExpensiTextInput extends Component {
+                                (hasError || errorText) && styles.borderColorDanger,
+                            ]}
+                        >
+                            {hasLabel ? (
+                                <ExpensiTextInputLabel
+                                    label={label}
+                                    labelTranslateX={
+                                        ignoreLabelTranslateX
+                                            ? new Animated.Value(0)
+                                            : this.state.labelTranslateX
+                                    }
+                                    labelTranslateY={this.state.labelTranslateY}
+                                    labelScale={this.state.labelScale}
+                                />
+                            ) : null}
+                            <TextInput
+                                ref={(ref) => {
+                                    if (typeof innerRef === 'function') { innerRef(ref); }
+                                    this.input = ref;
+                                }}
+                                // eslint-disable-next-line
+                                {...inputProps}
+                                value={value}
+                                placeholder={(this.state.isFocused || !label) ? placeholder : null}
+                                placeholderTextColor={themeColors.placeholderText}
+                                underlineColorAndroid="transparent"
+                                style={[...inputStyle, errorText ? styles.errorOutline : undefined]}
+                                onFocus={this.onFocus}
+                                onBlur={this.onBlur}
+                                onChangeText={this.setValue}
                             />
-                        ) : null}
-                        <TextInput
-                            ref={(ref) => {
-                                if (typeof innerRef === 'function') { innerRef(ref); }
-                                this.input = ref;
-                            }}
-                            // eslint-disable-next-line
-                            {...inputProps}
-                            value={value}
-                            placeholder={(this.state.isFocused || !label) ? placeholder : null}
-                            placeholderTextColor={themeColors.placeholderText}
-                            underlineColorAndroid="transparent"
-                            style={inputStyle}
-                            onFocus={this.onFocus}
-                            onBlur={this.onBlur}
-                            onChangeText={this.setValue}
-                        />
-                    </View>
-                </TouchableWithoutFeedback>
+                        </View>
+                    </TouchableWithoutFeedback>
+                </View>
+                {errorText !== '' && (
```

Comment:
Could errorText be undefined or null? Maybe we should just do `{errorText && (` to be safe?

Reply:
Honestly, we should be defining propTypes and defaultProps for this component, and default errorText to an empty string.

Reply:
You are right, errorText is sometimes `undefined` and sometimes `''`

Weirdly I get this error when I try `{errorText && (`:

![Screen Shot 2021-08-19 at 9 35 33 AM](https://user-images.githubusercontent.com/87341702/130108072-5c003e21-82e9-49e6-9692-145048986369.png)

Which is relates to this conversation: https://stackoverflow.com/a/53510213/16434681

Using `{Boolean(errorText) && (` to convert it to boolean seems to work without the error.

Reply:
defaultProps defined for errorText fix this, gonna resolve this conversation.